app/component-library/components-temp/ActionListItem/ActionListItem.tsx

```diff
@@ -0,0 +1,120 @@
+// Third party dependencies
+import React from 'react';
+import { Pressable } from 'react-native';
+
+// External dependencies
+import {
+  Box,
+  BoxAlignItems,
+  BoxFlexDirection,
+  BoxJustifyContent,
+  Icon,
+  IconSize,
+  Text as DesignSystemText,
+  TextColor,
+  TextVariant,
+  FontWeight,
+} from '@metamask/design-system-react-native';
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+
+// Internal dependencies
+import { ActionListItemProps } from './ActionListItem.types';
+
+const ActionListItem: React.FC<ActionListItemProps> = ({
+  label,
+  description,
+  startAccessory,
+  endAccessory,
+  iconName,
+  labelTextProps,
+  descriptionTextProps,
+  iconProps,
+  ...pressableProps
+}) => {
+  const tw = useTailwind();
+
+  // Render label based on type
+  const renderLabel = () => {
+    if (typeof label === 'string') {
+      return (
+        <DesignSystemText
+          variant={TextVariant.BodyMd}
+          fontWeight={FontWeight.Medium}
+          {...labelTextProps}
+        >
+          {label}
+        </DesignSystemText>
+      );
+    }
+    return label;
+  };
+
+  // Render description based on type
+  const renderDescription = () => {
+    if (!description) return null;
+
+    if (typeof description === 'string') {
+      return (
+        <DesignSystemText
+          variant={TextVariant.BodySm}
+          fontWeight={FontWeight.Medium}
+          color={TextColor.TextAlternative}
+          {...descriptionTextProps}
+        >
+          {description}
+        </DesignSystemText>
+      );
+    }
+    return description;
+  };
+
+  // Render start content (either startAccessory or icon)
+  const renderStartContent = () => {
+    if (startAccessory) {
+      return startAccessory;
+    }
+
+    if (iconName) {
+      return <Icon name={iconName} size={IconSize.Md} {...iconProps} />;
+    }
+
+    return null;
+  };
+
+  return (
+    <Pressable
+      style={({ pressed }) =>
+        tw.style('bg-default px-4 py-3', pressed && 'bg-default-pressed')
+      }
+      {...pressableProps}
+    >
+      <Box
+        flexDirection={BoxFlexDirection.Row}
+        alignItems={BoxAlignItems.Center}
+        justifyContent={BoxJustifyContent.Between}
+        twClassName="gap-4"
```

Comment:
non-blocking: should be able to use gap prop here
```suggestion
        gap={4}
```